python/tvm/topi/cuda/sparse.py

```diff
@@ -367,9 +367,14 @@ def _alter_sparse_dense_layout(_attrs, inputs, _tinfos, _out_type):
         and isinstance(inputs[2], relay.Constant)
         and isinstance(inputs[3], relay.Constant)
     ):
-        sparse_matrix = sp.bsr_matrix(
-            (inputs[1].data.asnumpy(), inputs[2].data.asnumpy(), inputs[3].data.asnumpy())
-        )
+        if len(inputs[1].data.asnumpy().shape) == 1:
```

Comment:
Actually the indices and ind pointers are in CSR format, so BSR_Matrix can not digest it in scipy. I got an error here.

Reply:
CSR format is just BSR format with block size one. You should be able to use the same indices and indptr as you do with CSR. What was there error you were getting?

Reply:
On paper, I totally agree with you. But when it comes to implementation, it differs always. For example, your data representation, csr.data & bsr.data has different way of representation. As i could remember, it was an error while deducing R & C parameter from csr.data, and that is quite expected, as the block size is not provided, it will deduce it from the input.

Reply:
Still confused on why this is necessary. The scipy docs show an example where a 1D data array is passed in to `bsr_matrix`. I guess it really doesn't matter though.

Reply:
I understand your confusion here.
I am refering the [link](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.bsr_matrix.html#scipy.sparse.bsr_matrix) .
As you can see the 1-D data is not only fed to bsr_matrix creation, also the coordinates (rows, cols) are provided alomg with shape. With this constructor , it internally forms indptrs & indices. But in our case, we have set of input as data(R x C) , indices & indptrs already, so we are utilizing different constructor. 

I hope it helps to resolve your query here. Thanks!

Reply:
It seems like the different constructors have different behavior. Interesting.

Thanks for clearing that up.